src/mesh/Mesh.cpp

```diff
@@ -466,5 +466,137 @@ std::ostream &operator<<(std::ostream &os, const Mesh &m)
+
+  /*
+  For the convex hull, the most left hand point regarding the x coordinate is chosen as the starting point. 
+  The algorithm moves in an anti-clockwise position, finding the most right hand coordinate from the 
+  previous most right hand point.
+  */
+  
+  //First find point with smallest x coord. This point must be in the convex set then and is the starting point of gift wrapping algorithm
+  int idLowestPoint = 0;
+  for (int i = 1; i < 4; i++) {
+    if (coords[i][0] < coords[idLowestPoint][0]){
+      idLowestPoint = i;
+    }
+  }
+
+  // Found starting point. Add this as the first vertex in the convex hull.
+  // current is the origin point => hull[0]
+  int validVertexIDCounter = 0;           // Counts number of times a valid vertex is found. Must be 4 for a valid quad.
+  int currentVertex = idLowestPoint;      // current valid vertex 
+  int nextVertex = 0;                     // Next potential valid vertex
+  do
+  {
+    // Add current point to result
+    hull[validVertexIDCounter]=currentVertex;
+    
+    nextVertex = (currentVertex + 1)%4;              // remainder resets loop through vector of points
+    for (int i = 0; i < 4; i++)
+    {
+      double y1 = coords[currentVertex][1] - coords[nextVertex][1];
+      double y2 = coords[currentVertex][1] - coords[i][1];
+      double x1 = coords[currentVertex][0] - coords[nextVertex][0];
+      double x2 = coords[currentVertex][0] - coords[i][0];
+      double val = y2 * x1 - y1 * x2;
+
+      if (val > 0){
+        nextVertex = i;
+      }
+    }
+    // Now nextVertex is the most anti-clockwise with respect to current
+    // Set current as nextVertex for next iteration, so that nextVertex is added to
+    // result 'hull'
+    currentVertex = nextVertex;
+    validVertexIDCounter++; 
+  } while (currentVertex != idLowestPoint);  // While we don't come to first point
+
+  if (validVertexIDCounter < 4){
+    //Error, quad is invalid
+    PRECICE_DEBUG("Invalid Quad. Hull: " << validVertexIDCounter);
+  } else {
+    PRECICE_DEBUG("Valid Quad. Hull: " << validVertexIDCounter);
+  }
+
+  //Ordering of quad is hull 0-1-2-3-0
+}
+
+void Mesh::computeQuadEdgeOrder(std::array<int,4> &edgeList, std::array<int,4> &vertexList) const
+{
+  /*
+  The first edge in the list is treated as the first edge (edge[0]). An edge that does not share a vertex
+  with the frist edge is the 3rd edge (edge[2]). The edge that shares the first vertex with the first
+  edge (shares vertex[0] with edge[0]) is the 4th edge (edge[3]) as long as it does not share
+  the other vertex with edge[0] (then there is a duplicated edge). The edge sharing vertex[1] and not
+  vertex[0] is the 2nd edge (edge[1]). This also provides the vertex ordering to determine the diagonal, 
+  but not convexity.
+  */
+
+  int edgeOrder[4];
+  //edgeOrder[0] = edgeList[0];   //Will order the edges in edgeList in order that they form a closed quad
```

Comment:
Do we need this or not? If not, let's not keep it here. Only confuses.

Reply:
Do you mean the commented line `//edgeOrder[0] = edgeList[0]`. I have removed it.